languages/elixir/exercises/concept/charlists/.docs/after.md

```diff
@@ -0,0 +1,164 @@
+Charlists are created using single quotes.
+
+```elixir
+'hello'
+```
+
+A charlist is a list of integers where the integers represent the Unicode values of a given character — also known as code points.
+
+## Charlists _are_ lists
+
+```elixir
+'' === []
+# => true
+
+is_list('hello')
+# => true
+```
+
+Because charlist are list, you can work with them just like with any other list - using recursion and pattern matching, or using the [`List`][list] module.
+
+```elixir
+[first_letter | _] = 'cat'
+first_letter
+# => 99
+
+List.first('hello')
+# => 104
+List.pop_at('hello', 0)
+# => {104, 'ello'}
+
+```
+
+You can concatenate two lists together using [`++`][kernel-concat-list].
+
+```elixir
+'hi' ++ '!'
+# => 'hi!'
+```
+
+The longer the first list is, the slower the concatenation, so avoid repeatedly appending to lists of arbitrary length.
+
+## Printability
+
+If a list of integers contains only integers that are code points of printable character, it will be displayed as a charlist. Even if it was defined using the `[]` syntax.
+
+```elixir
+'ABC'
+# => 'ABC'
+
+[65, 66, 67]
+# => 'ABC'
+```
+
+If a list of integers contains even one code point of an unprintable character (e.g. `0-6`, `14-26`, `28-31), it will be displayed as a list. Even if it was defined using the`''` syntax.
+
+```elixir
+'ABC\0'
+# => [65, 66, 67, 0]
+
+[65, 66, 67, 0]
+# => [65, 66, 67, 0]
+```
+
+Printability can be checked with [`List.ascii_printable?`].
+
+```elixir
+List.ascii_printable?([65, 66, 67])
+# => true
+List.ascii_printable?([65, 66, 67, 0])
+# => false
+```
+
+Keep in mind that those are two different ways of _displaying_ the same data. The values are strictly equal.
+
+```elixir
+'ABC' === [65, 66, 67]
+# => true
+```
+
+When printing a list with [`IO.inspect`][io-inspect], you can use the [`:charlists`][inspect-opts] option to control how lists are printed.
+
+```elixir
+IO.inspect('ABC', charlists: :as_charlists)
+# => prints 'ABC'
+
+IO.inspect('ABC', charlists: :as_lists)
```

Comment:
Since we have refrained from talking much about IO, should this be adapted to `Kernel.inspect` rather than `IO.inspect`?

Reply:
I understand where you're coming from. I like that this would mean we don't have to write "prints ..." because `Kernel.inspect` just returns the value. On the other hand, I am worried that people will be confused by the fact that the return values are strings. I'm not sure what's better...
```elixir
inspect('ABC', charlists: :as_charlists)
# => "'ABC'"
```

Do we have any plans for teaching about `IO`?

Reply:
Hmm. Good point. No I don't think there are plans yet for IO. Maybe keeping it as IO.inspect is the better choice if we are going to mention it at all.

Reply:
IO.inspect / IO.puts calls will be caught by the test runner and returned to the student, so it might be the better thing to have if going to have an example.